app/api/schema/discount_codes.py

```diff
@@ -117,7 +117,7 @@ def validate_date(self, data, original_data):
                 data['valid_from'] = discount_code.valid_from
 
             if 'valid_till' not in data:
-                data['valid_till'] = discount_code.valid_till
+                data['valid_till'] = discount_code.valid_expire_time
```

Comment:
This defeats the entire purpose of the change. Now, it will set to event ends at and be static instead of dynamic